src/javascript/app_2/Stores/Modules/Contract/contract-store.js

```diff
@@ -53,9 +56,16 @@ export default class ContractStore extends BaseStore {
 
         if (end_time) {
             SmartChartStore.setRange(contract_info.date_start, end_time);
+
+            if (!contract_info.tick_count && !this.is_granularity_set) {
```

Comment:
maybe we could put this as a `should_update_chart_type` or similar? + use on line 67 since it's the same check